extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/InterfaceConfigPropertiesUtil.java

```diff
@@ -196,26 +199,33 @@ private static boolean isDefault(short flags) {
         return ((flags & (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC)) == Modifier.PUBLIC);
     }
 
-    private static NameAndDefaultValue determinePropertyNameAndDefaultValue(MethodInfo method) {
+    private static NameAndDefaultValue determinePropertyNameAndDefaultValue(MethodInfo method,
+            ConfigProperties.NamingStrategy namingStrategy) {
         AnnotationInstance configPropertyAnnotation = method.annotation(DotNames.CONFIG_PROPERTY);
         if (configPropertyAnnotation != null) {
             AnnotationValue nameValue = configPropertyAnnotation.value("name");
-            String name = (nameValue == null) || nameValue.asString().isEmpty() ? getPropertyNameFromMethodName(method)
+            String name = (nameValue == null) || nameValue.asString().isEmpty() ? getPropertyName(method, namingStrategy)
                     : nameValue.asString();
             AnnotationValue defaultValue = configPropertyAnnotation.value("defaultValue");
 
             return new NameAndDefaultValue(name, defaultValue != null ? defaultValue.asString() : null);
         }
 
-        return new NameAndDefaultValue(getPropertyNameFromMethodName(method));
+        return new NameAndDefaultValue(getPropertyName(method, namingStrategy));
     }
 
-    private static String getPropertyNameFromMethodName(MethodInfo method) {
-        try {
-            return JavaBeanUtil.getPropertyNameFromGetter(method.name());
-        } catch (IllegalArgumentException e) {
-            throw new IllegalArgumentException("Method " + method.name() + " of interface " + method.declaringClass()
-                    + " is not a getter method. Either rename the method to follow getter name conventions or annotate the method with @ConfigProperty");
+    private static String getPropertyName(MethodInfo method, ConfigProperties.NamingStrategy namingStrategy) {
+        if (namingStrategy == ConfigProperties.NamingStrategy.DEFAULT) {
+            try {
+                return JavaBeanUtil.getPropertyNameFromGetter(method.name());
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Method " + method.name() + " of interface " + method.declaringClass()
+                        + " is not a getter method. Either rename the method to follow getter name conventions or annotate the method with @ConfigProperty");
+            }
+        } else if (namingStrategy == ConfigProperties.NamingStrategy.KEBAB_CASE) {
+            return StringUtil.hyphenate(method.name());
```

Comment:
How about checking for a getter in here the same as in the default case? Looking at the original issue (https://github.com/quarkusio/quarkus/issues/6457), this seems to be the expected scenario i.e. `schema-location` > `getSchemaLocation`

Reply:
Good idea

Reply:
Done